### PR TITLE
Muzzle servlet-3 Instrumentation

### PR DESCRIPTION
**Pull request by gary-huang**
Add muzzle block and remove class loader matcher method.

### CODE REVIEW

**Comment by realark** [comment]
Does this contradict the version ranges in the servlet 2 integration? https://github.com/DataDog/dd-trace-java/pull/483

File: dd-java-agent/instrumentation/servlet-3/servlet-3.gradle [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
-  legacyModule = "servlet-api"
-  versions = "[3.0,)"
-  verifyPresent = [
-    "javax.servlet.AsyncEvent"   : null,
-    "javax.servlet.AsyncListener": null,
-  ]
+muzzle {
+  pass {
+    group = "javax.servlet"
+    module = 'javax.servlet-api'
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/servlet-3/servlet-3.gradle]: This has a legacy module name (`servlet-api`).  You should add an extra block to cover those.

File: dd-java-agent/instrumentation/servlet-3/servlet-3.gradle [hunk 2]
```diff
@@ -5,6 +5,11 @@ muzzle {
     versions = "[3.0,)"
     assertInverse = true
   }
+  fail {
+    group = "javax.servlet"
+    module = 'servlet-api'
+    versions = "[2.3,)"
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/servlet-3/servlet-3.gradle]: probably don't need to specify the lower bounds here.

File: dd-java-agent/instrumentation/servlet-3/servlet-3.gradle [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+  pass {
+    group = "javax.servlet"
+    module = 'javax.servlet-api'
+    versions = "[3.0,)"
+    assertInverse = true
+  }
+  fail {
+    group = "javax.servlet"
+    module = 'servlet-api'
+    versions = "[,)"
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/servlet-3/servlet-3.gradle]: nit.  please change to `(,)`

**Pull request merged**